Source/WarpX.cpp

```diff
@@ -1176,52 +1178,57 @@ WarpX::ReadParameters ()
-        if (m_v_galilean[0] != 0. || m_v_galilean[1] != 0. || m_v_galilean[2] != 0.)
-        {
-            if (m_v_comoving[0] != 0. || m_v_comoving[1] != 0. || m_v_comoving[2] != 0.)
-            {
-                amrex::Abort("Galilean and comoving algorithms should not be used together");
-            }
-        }
-
         // Scale the Galilean/comoving velocity by the speed of light
-        for (int i=0; i<3; i++) m_v_galilean[i] *= PhysConst::c;
-        for (int i=0; i<3; i++) m_v_comoving[i] *= PhysConst::c;
+        for (auto& vv : m_v_galilean) vv*= PhysConst::c;
+        for (auto& vv : m_v_comoving) vv*= PhysConst::c;
+
+        const auto v_galilean_is_zero =
+            std::all_of(m_v_galilean.begin(), m_v_galilean.end(),
+            [](const auto& val){return val == 0.;});
+
+        const auto v_comoving_is_zero =
+            std::all_of(m_v_comoving.begin(), m_v_comoving.end(),
+            [](const auto& val){return val == 0.;});
+
+
+        // Galilean and comoving algorithms should not be used together
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+            v_galilean_is_zero || v_comoving_is_zero,
+            "Galilean and comoving algorithms should not be used together"
+        );
 
         // The comoving PSATD algorithm is not implemented nor tested with Esirkepov current deposition
-        if (current_deposition_algo == CurrentDepositionAlgo::Esirkepov) {
-            if (m_v_comoving[0] != 0. || m_v_comoving[1] != 0. || m_v_comoving[2] != 0.) {
-                amrex::Abort("Esirkepov current deposition cannot be used with the comoving PSATD algorithm");
-            }
-        }
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+            (current_deposition_algo != CurrentDepositionAlgo::Esirkepov) ||
+            v_comoving_is_zero,
+            "Esirkepov current deposition cannot be used with the comoving PSATD algorithm"
+        );
 
-        if (current_deposition_algo == CurrentDepositionAlgo::Vay) {
-            if (m_v_galilean[0] != 0. || m_v_galilean[1] != 0. || m_v_galilean[2] != 0.) {
-                amrex::Abort("Vay current deposition not implemented for Galilean algorithms");
-            }
-        }
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+            (current_deposition_algo != CurrentDepositionAlgo::Vay) ||
+            v_galilean_is_zero,
+            "Vay current deposition not implemented for Galilean algorithms"
+        );
 
-        if (current_correction) {
-            if (m_v_galilean[0] != 0. || m_v_galilean[1] != 0. || m_v_galilean[2] != 0.) {
-                if (fft_do_time_averaging) {
-                    amrex::Abort("Current correction not implemented for averaged Galilean algorithm");
-                }
-            }
-        }
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+            (!current_correction) || v_galilean_is_zero || (!fft_do_time_averaging),
+            "Current correction not implemented for averaged Galilean algorithm"
```

Comment:
I don't think this assert is correct yet, but too late rn to get the triple if right :sweat_smile: 

memo to myself: check from here to end tomorrow

Reply:
I actually think it is! `if (A and B and C) {ABORT}` means that it is enough that either not(A) or not(B) or not(C) are True to prevent the abort!